### PR TITLE
--set-prefix option works for old loop

### PR DESCRIPTION
**Pull request by robertoleksy**

### CODE REVIEW

File: src/tunserver.cpp [hunk 1]
```diff
@@ -1906,6 +1907,10 @@ void c_tunserver::set_remove_peer_tometout(unsigned int timeout_seconds) {
 	peer_timeout = std::chrono::seconds(timeout_seconds);
 }
 
+void c_tunserver::set_prefix_len(int prefix) {
```
Comment (woskov) [inline src/tunserver.cpp]: inline maybe

File: src/tunserver.cpp [hunk 2]
```diff
@@ -609,6 +611,7 @@ void c_tunserver::configure_mykey() {
 
 	// save signature and IDI publickey in tunserver
 	m_my_IDI_pub = my_IDI->m_pub;
+	m_my_IDI_pub.set_ipv6_prefix(m_ipv6_prefix);
```
Comment (rfree-d) [inline src/tunserver.cpp]: This is not a property of crypto key, what prefix/mask it has...

We need to rethink this all.

But can be ok for now to test how internal network works via galaxy42.

File: src/tunserver.cpp [hunk 3]
```diff
@@ -1906,6 +1913,14 @@ void c_tunserver::set_remove_peer_tometout(unsigned int timeout_seconds) {
 	peer_timeout = std::chrono::seconds(timeout_seconds);
 }
 
+void c_tunserver::set_prefix_len(int prefix) {
+	m_prefix_len = prefix;
+}
+
+void c_tunserver::set_prefix(const string &prefix) {
+	m_ipv6_prefix = prefix;
```
Comment (rfree-d) [inline src/tunserver.cpp]: @robertoleksy this var naming is not consistent, imo better

m_ipv6_prefix_addr
m_ipv6_prefix_len 

please apply even when this is merged

File: src/tunserver.hpp [hunk 4]
```diff
@@ -236,6 +236,8 @@ class c_tunserver : public c_galaxy_node {
 		int get_ip_protocol_number(const std::string& data) const;
 		void enable_remove_peers();
 		void set_remove_peer_tometout(unsigned int timeout_seconds);
+		void set_prefix_len(int prefix);
+		void set_prefix(const std::string &prefix);
```
Comment (rfree-d) [inline src/tunserver.hpp]: also function name, like vars

set_ipv6_prefix_addr
set_ipv6_prefix_len

**Pull request merged**